pkg/clients/openshift/healthcheck.go

```diff
@@ -0,0 +1,106 @@
+package openshift
```

Comment:
Are you looking to remove this from the openshift clients package and put it directly alongside the cluster type (e.g. osd, rosa, rosa hcp) or do you mean to just have 3 separate methods within this file for each cluster type?

It would be great at some point if the osd-cluster-ready job can support hosted control plane clusters. I know there is one limitation with not knowing if the cluster is a hosted control plane cluster or a standard managed openshift cluster. Once this is supported, I feel this would solve a lot of our health check code and simplify it.

Reply:
I opened _\SDCICD-1043_ where we can track this suggestion and implement it in a separate PR. This way we can move this initial PR along.